lib/lib-doi/src/main/java/org/sagebionetworks/doi/datacite/DataciteClientImpl.java

```diff
@@ -0,0 +1,209 @@
+package org.sagebionetworks.doi.datacite;
+
+
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.sagebionetworks.repo.model.doi.v2.DataciteMetadata;
+import org.sagebionetworks.repo.web.ForbiddenException;
+import org.sagebionetworks.repo.web.NotFoundException;
+import org.sagebionetworks.repo.web.ServiceUnavailableException;
+import org.sagebionetworks.simpleHttpClient.*;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DataciteClientImpl implements DataciteClient {
+
+	@Autowired
+	private DataciteMetadataTranslator metadataTranslator;
+
+	@Autowired
+	private DataciteXmlTranslator xmlTranslator;
+
+	private static final Integer TIME_OUT = 9000; // 9 seconds
+	private static final String USER_AGENT = "Synapse";
+	private String DATACITE_URL;
+	private String USERNAME;
+	private String PASSWORD;
+	private final SimpleHttpClient client;
+
+	public DataciteClientImpl() {
+		// Configure HTTP client for use
+		SimpleHttpClientConfig httpClientConfig = new SimpleHttpClientConfig();
+		httpClientConfig.setSocketTimeoutMs(TIME_OUT);
+		client = new SimpleHttpClientImpl(httpClientConfig);
+	}
+
+	public void setConfig(DataciteClientConfig config) {
+		USERNAME = config.getUsername();
+		PASSWORD = config.getPassword();
+		DATACITE_URL = config.getDataciteUrl();
+	}
+
+	@Override
+	public DataciteMetadata get(final String doiUri) throws ServiceUnavailableException {
+		// Call Datacite API to get metadata
+		SimpleHttpRequest request = new SimpleHttpRequest();
+		request.setUri(DATACITE_URL + "metadata/" + doiUri);
+		Map<String, String> headers = new HashMap<>();
+		headers.put(HttpHeaders.AUTHORIZATION, USERNAME + ":" + PASSWORD);
+		headers.put(HttpHeaders.USER_AGENT, USER_AGENT);
+		request.setHeaders(headers);
+
+		SimpleHttpResponse response = null;
+		try {
+			response = client.get(request);
+		} catch (IOException e) {
+			throw new ServiceUnavailableException(e);
+		}
+
+		// See Datacite MDS support docs: https://support.datacite.org/docs/mds-api-guide#section-getting-started
+		// Outlines all possible HTTP response status codes
+		String xml = null;
+		if (response.getStatusCode() == HttpStatus.SC_OK) { // Success
+			xml = response.getContent();
+		} else if (response.getStatusCode() == HttpStatus.SC_NO_CONTENT) { // Not minted or not resolvable
+			throw new NotFoundException("DOI " + doiUri + " was not minted in DataCite or was not resolvable.");
+		} else if (response.getStatusCode() == HttpStatus.SC_UNAUTHORIZED) { // No login - this is an issue with us or DataCite
+			throw new ServiceUnavailableException("Error authenticating with DataCite.");
```

Comment:
Needs to result in 500 (runtime?) as there is nothing for the user to do except report the failure to support.